ci/scripts/nanoarrow_build.sh

```diff
@@ -27,18 +27,21 @@ build_dir=${2}/nanoarrow
 # integration tests. Testing of the nanoarrow implementation in normal CI is handled
 # by github workflows in the arrow-nanoarrow repository.
 
+# Include in the integration test by default if the source directory is present
```

Comment:
The problem is that the flag is still off by default in Archery:
https://github.com/apache/arrow/blob/680980e8bd70393e95f65fa880914aa6c32aedaa/dev/archery/archery/cli.py#L741-L743